native_client/tfmodelstate.cc

```diff
@@ -50,17 +49,19 @@ TFModelState::init(const char* model_path,
     options.config.mutable_graph_options()
       ->mutable_optimizer_options()
       ->set_opt_level(::OptimizerOptions::L0);
-    options.env = mmap_env_;
+    options.env = mmap_env_.get();
   }
 
-  status = NewSession(options, &session_);
+  Session* _session;
+  status = NewSession(options, &_session);
   if (!status.ok()) {
     std::cerr << status << std::endl;
     return DS_ERR_FAIL_INIT_SESS;
   }
+  session_ = std::unique_ptr<tensorflow::Session>(_session);
```

Comment:
`session_.reset(session);`

Reply:
Always forget about `.reset()`

Reply:
We could also `.reset()` for `mmap_env_` :)